### PR TITLE
Add Fiber#kill, similar to Thread#kill.

### PR DESCRIPTION
**Pull request by ioquatix**
https://bugs.ruby-lang.org/issues/595

### CODE REVIEW

File: spec/ruby/core/fiber/kill_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    it "kills a resumed fiber" do
+      fiber = Fiber.new{while true; Fiber.yield; end}
+      fiber.resume
+
+      fiber.alive?.should == true
+
+      fiber.kill
+      fiber.alive?.should == false
+    end
+
```
Comment (nobu) [inline spec/ruby/core/fiber/kill_spec.rb:25]:
```suggestion

```

**Comment by hsbt** [comment]
@ioquatix Do not create branch on `ruby/ruby`.

**Comment by ioquatix** [comment]
Ah sorry I must have pushed to the wrong repo! I'll delete it.

**Comment by ioquatix** [comment]
I will re-open a new PR later.

**Comment by hsbt** [comment]
Thanks ❤️

**Pull request closed without merging**